File: app/src/main/java/ru/mamykin/foreignbooksreader/views/DeviceBooksView.java
package ru.mamykin.foreignbooksreader.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

import ru.mamykin.foreignbooksreader.models.AndroidFile;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface DeviceBooksView extends MvpView {
    void showFiles(List<AndroidFile> files);

    void setCurrentDir(String currentDir);

    void showPermissionMessage();

    void openBook(String path);

    void showUpDir(boolean show);
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/presentation/device/DeviceBooksView.kt
package ru.mamykin.foreignbooksreader.presentation.device

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

import ru.mamykin.foreignbooksreader.models.AndroidFile

@StateStrategyType(AddToEndSingleStrategy::class)
interface DeviceBooksView : MvpView {
    fun showFiles(files: List<AndroidFile>)

    fun setCurrentDir(currentDir: String)

    fun showPermissionMessage()

    fun openBook(path: String)

    fun showUpDir(show: Boolean)
}